### DISCUSSION TITLE
Is that possible to exclude a directory from recursive install?

### DISCUSSION THREAD

**Discussion by fa93hws** [pnpm/pnpm#7992] [Q&A]
Hi pnpm friends, just wondering is there a way to exclude one directory from `pnpm install --recursive`?
For example, in my `pnpm-workspace.yml`, there is 
```
foo
bar
foobar
```
and I would like **not** to install `node_modules` for `foobar` when running `pnpm install --recursive` in `foo` or `bar`, does anyone know whether it's possible?

**For context:**
I'm migrating our mono-repo to pnpm from yarn. After migrating most of the nodejs packages to pnpm, there is one giant package left which serves the most important product in my company and there are more than 300 engineers work with it in a daily basis. Meaning I can't take the risk to migrate it in one go. As a consequence, I plan to modify the [yarnrc:yarn-path](https://classic.yarnpkg.com/lang/en/docs/yarnrc/#toc-yarn-path) so that when `EXPERIMENTAL_PNPM_AS_YARN` is `true`, `pnpm` will be invoked even though `yarn` is used. e.g. in yarnrc file:
``` 
  if [[ "${EXPERIMENTAL_PNPM_AS_YARN:-}" == "true" ]]; then
    exec pnpm "$@"
  else
    exec yarn "$@"
  fi
```
By doing so, I'm able to keep the CI the same while asking people to opt-in to `pnpm` progressively, and flip the flag when it's proven to be safe.
However, one of the problem in this approach is that when people running `pnpm` in some other packages, `node_modules` for this giant package will be installed as well and break their `yarn` experience.

**Comment by ayZagen** [comment]
To answer your question, have a look at [pnpm filtering](https://pnpm.io/filtering), specifically to [excluding section](https://pnpm.io/filtering#excluding)

For your use case though, I don't think it is a good idea as pnpm and yarn does lots of things differently and you may need to configure pnpm accordingly.

**Reply by fa93hws** [reply]
Hi many thanks for the reply, yea filtering works but it's not ideal in my use case, because it's inpossible to ask people to always run `pnpm --filter xxx install` in other packages.

Is there any existing example for migrating large package from yarn to pnpm?
